src/app.js

```diff
@@ -1,24 +1,137 @@
+
+const env = require("./configs/index");
+const logger = require("./middleware/logger");
+const userRoutes = require("./routes/users");
+const actionStepsRoutes = require("./routes/actionsteps");
+const weightRoutes = require("./routes/weight");
+const categoryRoutes = require("./routes/category");
+const potentialRoutes = require("./routes/potential");
+const costRoutes = require("./routes/cost");
+const timescaleRoutes = require("./routes/timescale");
+
+const url = process.env.MONGO_URL;
+
+const connect = mongoose.connect(url);
+connect.then(
+  () => {
+    console.log("connected Correctly");
+  },
+  (err) => {
+    console.error(err);
+  }
+);
+app.use(
+  expresssession({
+    secret: process.env.SECRET_KEY,
+    resave: false,
+    saveUninitialized: true,
+    store: new MongoStore({ mongoUrl: url }),
+  })
+);
+
+app.use(cors({ origin: "*" }));
+app.use(express.static("./assets"));
+app.use(express.static(path.join(__dirname, "public")));
+app.use(express.json({ extended: true }));
+app.use(express.urlencoded({ extended: true }));
+
+// app.use(responseTime);
+
+const swaggerOptions = {
+  definition: {
+    openapi: "3.0.0",
+    info: {
+      title: "Green-Me Official",
+      version: "0.1.0",
+      description: "API Routes and Database Schema for GreenMe-Official",
+    },
+    servers: [
+      {
+        url: `http://${process.env.HOST}/${process.env.PORT}`,
+      },
+    ],
+  },
+  apis: ["src/routes/actionsteps.js"],
+};
+const swaggerSpec = swaggerJSDoc(swaggerOptions);
+app.use(
+  "/api-docs",
+  swaggerUi.serve,
+  swaggerUi.setup(swaggerSpec, { explorer: true })
+);
+
+app.get("/api-docs.json", (req, res) => {
+  res.setHeader("Content-Type", "application/json");
+  res.send(swaggerSpec);
+});
+app.use((req, res, next) => {
+  // Log the request
+  logger.info(`[${req.method}] ${req.originalUrl}`);
+  next();
+});
 app.use("/api/v1/healthcheck", healthcheckRoute);
 app.use("/api/v1/users", userRoutes);
+app.use("/api/v1/actionsteps", actionStepsRoutes);
+app.use("/api/v1/weights", weightRoutes);
+app.use("/api/v1/categories", categoryRoutes);
+app.use("/api/v1/potentials", potentialRoutes);
+app.use("/api/v1/costs", costRoutes);
+app.use("/api/v1/timescale", timescaleRoutes);
+
+app.use((req, res, next) => {
+  const err = new Error();
+  err.status = 404;
+  err.message = "Route not found";
+  next(err);
+});
+
+app.use((err, req, res, next) => {
+  logger.error(err.stack);
+
+  res.status(err.status || 500).json({
+    success: false,
+    message: err.message || "internal server error",
+  });
+  next();
+});
+
+// error handler
+// eslint-disable-next-line no-unused-vars
+app.use((err, req, res, next) => {
+  let status = err.status || 500;
+  let message = err.message || "Internal Server Error";
+
+  if (err.name === "MongoServerError" && err.code === 11000) {
+    status = 400;
+    message = "Duplicate key error";
+  } else if (err.name === "ValidationError") {
```

Comment:
This error is handled in the error-handler middleware. No point duplicating it.

Reply:
fixed